### DISCUSSION TITLE
4.2 隐式序列

### DISCUSSION THREAD

**Discussion by giscus** [csfive/composing-programs-zh#96] [Announcements]
# 4.2 隐式序列

SICP Python 中文翻译

https://composingprograms.netlify.app/4/2

**Comment by CharlieCRX** [comment]
这节也不简单啊😑感谢翻译大大

**Comment by JzzzX** [comment]
补充一个知乎的yield，这一节的重点应该就是yield生成器 https://zhuanlan.zhihu.com/p/268605982

**Reply by JzzzX** [reply]
官方文档中的生成器内容——https://docs.python.org/zh-cn/3/reference/expressions.html#yield-expressions

**Comment by enternal111** [comment]
map_stream函数返回表达式里应该是fn(s.first)，写成逗号了

**Reply by enternal111** [reply]
很多函数代码和原文有出入，少了return之类的，建议看原文

**Comment by jackyang714** [comment]
感谢大佬翻译

**Comment by lemonacidl** [comment]
这节课的重点应该是惰性计算，迭代器和后面的Python流看不出有很大关系

**Reply by haoaaa-111** [reply]
这个应该是一个东西啊 惰性计算是一种思想 迭代器和流是惰性计算的具体形式

**Comment by haoaaa-111** [comment]
我学了之后感觉这一节里面最难的那个stream 如果把他看成一种特殊的scheme列表形式去理解会好很多
